relayer/src/worker/packet.rs

```diff
@@ -129,18 +123,18 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> PacketWorker<ChainA, ChainB> {
                     height,
                     new_block: _,
                 } => {
-                    let (first_run, clear_on_start) = {
-                        match self.clear_packets_on_start.take() {
-                            None => (false, false),
-                            Some(clear_on_start) => (true, clear_on_start),
-                        }
+                    let clear_on_start = if self.packets_cfg.clear_on_start {
+                        self.packets_cfg.clear_on_start = false;
```

Comment:
I'd rather we maintain a `first_run` variable or field somewhere to make this behavior a bit more explicit if that makes sense?

Reply:
https://github.com/informalsystems/ibc-rs/pull/1539/commits/891251130a421c9146182f30fb399cee5367a53e